### PR TITLE
Fix #1181 Update MasterMetadataUtil to use new Ample api

### LINKED ISSUE #1181: Update MasterMetadataUtil.addNewTablet to use new Ample API
The method `MasterMetadataUtil.addNewTablet()` could be updated to use the new Ample API (an internal API for reading and writing Accumulo metadata).

### PR DESCRIPTION
**Pull request by hkeebler**
@keith-turner , I pushed this for review.  It's pretty straightforward.  the sunny tests ran and I ran many of the Split IT tests that seemed to use it.  In process of testing against the standalone instance I ran into problems and need to recreate my whole setup and will provide a status on that when its working.

Note my comment on the collections assumed "not null".  Personally I dislike null checks and usually try to ensure objects floating around are not null  and have noticed they are not prevalent in the code base BUT, often the checks do serve a needed purpose.

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java [hunk 1]
```diff
@@ -151,6 +166,20 @@ private String getLocationFamily(LocationType type) {
     return this;
   }
 
+  @Override
+  public Ample.TabletMutator putBulkFile(Ample.FileMeta bulkref, long tid) {
+    Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
+    Value tidBytes = new Value(Long.toString(tid).getBytes());
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: Should use the Value constructor that takes String

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java [hunk 2]
```diff
@@ -151,6 +166,20 @@ private String getLocationFamily(LocationType type) {
...
+    mutation.put(TabletsSection.BulkFileColumnFamily.NAME, bulkref.meta(), tidBytes);
+    return this;
+  }
+
+  // not sure what to do here yet
+  @Override
+  public Ample.TabletMutator deleteBulkFile(Ample.FileMeta bulkref) {
+    throw new UnsupportedOperationException();
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: Probably the following

```java
Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
mutation.putDelete(TabletsSection.BulkFileColumnFamily.NAME, bulkref.meta());
```

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java [hunk 3]
```diff
@@ -95,6 +95,21 @@ protected TabletMutatorBase(ServerContext ctx, KeyExtent extent) {
...
+  @Override
+  public Ample.TabletMutator putFlushId(long flushId) {
+    Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
+    TabletsSection.ServerColumnFamily.FLUSH_COLUMN.put(mutation,
+        new Value(("" + flushId).getBytes()));
+    return this;
+  }
+
+  @Override
+  public Ample.TabletMutator putTime(String time) {
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: I thought of #1210 while looking at this.  Using String is fine for this PR as adding the type would be a much bigger and unrelated change.

**Comment by keith-turner** [comment]
>  Personally I dislike null checks and usually try to ensure objects floating around are not null and have noticed they are not prevalent in the code base BUT, often the checks do serve a needed purpose.

I like them when it causes fail fast.  For example the following code would not fail fast.  Adding a null check would fail fast.

```java
void setItems(List items) {
  // If null, this code may fail later when something tries to use this.item.. would be nicer to get 
  // a stacktrace for the code calling set
  this.items = items;
}
```

If the function is using the passed in argument a NPE would naturally happen w/o an explicit null check.

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java [hunk 4]
```diff
@@ -95,6 +95,21 @@ protected TabletMutatorBase(ServerContext ctx, KeyExtent extent) {
...
+    Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
+    TabletsSection.ServerColumnFamily.TIME_COLUMN.put(mutation, new Value(time.getBytes()));
```
[the remark this answers is not in the document]
Comment (ctubbsii) [reply server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: :+1: For these, I like to do a static import of `StandardCharsets.UTF_8`, so you can just do `.getBytes(UTF_8)`

File: server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java [hunk 5]
```diff
@@ -69,34 +69,35 @@ public static void addNewTablet(ServerContext context, KeyExtent extent, String
...
 
     if (location != null) {
-      location.putLocation(m);
-      location.clearFutureLocation(m);
+      mutator.putLocation(location, LocationType.CURRENT);
+      mutator.deleteLocation(location, LocationType.FUTURE);
     }
-
+    // are the following collections assumed not null? at least any empty collection is passed?
     for (Entry<FileRef,DataFileValue> entry : datafileSizes.entrySet()) {
```
[the remark this answers is not in the document]
Comment (hkeebler) [reply server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java]: I'm less familiar with this newer java construct  but am making this change because I love one-liner code although I do think it is less obvious as to what is going on here - as to what information is being stored in the metadata table.

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (hkeebler) [reply server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: I'm cleaning up all of these to use the String (or CharSequence) version of Value() based on 
@keith-turner recommendation.

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java (hunk 2 above)
Comment (hkeebler) [reply server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: Added this although nothing out there uses it at this time.

File: server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java (hunk 3 above)
Comment (hkeebler) [reply server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java]: Yes,  a time object would be good.  We can update later.

File: server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (keith-turner) [reply server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java]: I read the following when I was learning about method references.

https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html

[the remark this answers is not in the document]
Comment (hkeebler) [reply server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java]: thanks.

**Comment by hkeebler** [comment]
commit includes suggested changes and java 11 testing, includes some standalone testing.
